SECURITY.md

```diff
@@ -4,11 +4,15 @@
 
 The versions of the project that are currently supported with security updates.
 
-| Version | Supported          |
-| ------: | :----------------- |
-|     6.x | :white_check_mark: |
-|     5.x | :white_check_mark: |
-|   < 5.0 | :x:                |
+| MUI X version | Release    | Supported                            |
+| ------------: | :--------- | :----------------------------------- |
+|        ^7.0.0 | 2024-03-23 | :white_check_mark: Stable major      |
+|        ^6.0.0 | 2023-03-03 | :white_check_mark: Long-term support |
+|        ^5.0.0 | 2021-11-23 | :x:                                  |
+|        ^4.0.0 | 2021-09-28 | :x:                                  |
+|        ^3.0.0 | /          | :x:                                  |
+|        ^2.0.0 | /          | :x:                                  |
+|        ^1.0.0 | /          | :x:                                  |
```

Comment:
Does it make sense to list these when the first released version was `4.0.0`? 🤔

Reply:
Ok, let's keep this change smaller in scope, removing this.